lc-0226-invert-binary-tree/src/test/java/solution/CustomCsvArgTest.java

```diff
@@ -0,0 +1,190 @@
+                    values.add(null);
+                }
+            }
+        }
+
+        // Generate List from TreeNode head:
+        MyTreeAsList(TreeNode head) {
+
+            values = new ArrayList<>();
+
+            if (head == null) {
+                values.add(null); // Add a 'null' instead of having empty List to stay consistent
+            } else {
+                Queue<TreeNode> q = new LinkedList<>();
+                q.offer(head);
+
+                // Build arraylist:
+                while (!q.isEmpty()) {
+                    TreeNode curr = q.poll();
+                    if (curr != null) {
+                        values.add(curr.val);
+                        q.offer(curr.left);
+                        q.offer(curr.right);
+                    } else {
+                        values.add(null);
+                    }
+                }
+
+                // Trim tail nulls:
+                while (values.get(values.size() - 1) == null) {
+                    values.remove(values.size() - 1);
+                }
+            }
+        }
+    }
+
+    static class TestTreeNode extends TreeNode {
+        private TestTreeNode(List<Integer> values) {
+            super(); // Instantiate an empty 'this' node
+
+            int values_size = values.size();
+
+            // Initialize head:
+            if (values_size > 0 && values.get(0) != null) {
+                this.val = values.get(0);
+            }
+
+            // Build a queue to build the tree:
+            Queue<TreeNode> q = new LinkedList<>();
+            q.offer(this);
+
+            int i = 1;
+            while (i < values_size) {
+
+                TreeNode curr = q.poll();
+
+                // If the current value is null, it can't have children:
+                if (curr == null) {
+                    i = i + 2;
+                } else {
+                    if (i < values_size && values.get(i) != null) {
+                        curr.left = new TreeNode(values.get(i));
+                        q.offer(curr.left);
+                    }
+                    i++;
+
+                    if (i < values_size && values.get(i) != null) {
+                        curr.right = new TreeNode(values.get(i));
+                        q.offer(curr.right);
+                    }
+                    i++;
+                }
+            }
+        }
+
+        // Factory method to control 'null' and empty string inputs. If constructed
+        // directly, 'super()' will instantiate a node with 'val = 0', instead of not
+        // generating a node at all. This way we can return a 'null' 'TreeNode' when the
+        // inputs are 'null':
+        static TestTreeNode fromString(String s) {
+            List<Integer> values = new MyTreeAsList(s).values;
+
+            if (values.isEmpty() || values.get(0) == null) {
+                return null;
+            } else {
+                return new TestTreeNode(values);
+            }
+        }
```

Comment:
Response to your original [comment](https://github.com/nekotuko/problem-solving/pull/14/files#r1367799714):

This was my approach based on your comment. I didn't want to change the original class ``TreeNode``, as defined in LeetCode. Instead, I implemented a ``TestTreeNode`` that inherits ``TreeNode``, but also uses a factory method ``fromString()`` to avoid instantiating ``null`` inputs. As a result:

1) The ``TestTreeNode`` class inherits ``TreeNode``, and constructs the Tree from input, without the need of a ``head`` pointer.
2) The ``fromString()`` factory method controls the conditions of instantiation, which in this case is to avoid instantiating a ``null`` node.